### ISSUE TITLE
[regression/8.0.0-rc.1.9171] RefreshView indicator hidden behind Navigation bar

### ISSUE THREAD

**Issue by KarsaOrlong1981** [dotnet/maui#17610]
### Description

When using an RefreshView at .net 8 the indicator at refreshing is always hidden behind navigation bar, before at .net 7 I can see it completly.

.net 8
![behind](https://github.com/dotnet/maui/assets/69248804/564277e8-52e1-4a66-a972-9379d125d03c)

.net 7
![behind7](https://github.com/dotnet/maui/assets/69248804/5cc7dc03-0d3b-4f39-8f0d-7014a08f1374)

```  
<ContentPage.Content>
     <RefreshView  >
             <CollectionView 
                  ItemsSource="{Binding Employees}" 
                  ItemTemplate="{StaticResource EmployeeTemplate}" 
                  ItemsLayout="VerticalGrid, 2" />
         </RefreshView>
        

     
 </ContentPage.Content>
 ```

### Version with bug

[8.0.0-rc.1.9171](https://github.com/dotnet/maui/releases/tag/8.0.0-rc.1.9171)

### Is this a regression from previous behavior?

Yes, this used to work in .NET MAUI

### Last version that worked well

[8.0.0-preview.7.8842](https://github.com/dotnet/maui/releases/tag/8.0.0-preview.7.8842)

### Affected platforms

Android

### Affected platform versions

Android 13

**Comment by andyx48** [comment]
I'm having the same issue with the version 8.0.0-rc.1.9171

After some investigation i think [this commit](https://github.com/dotnet/maui/pull/17080/commits/f9fb2ba92e606397ec10b718d7cd21bafc2a4cd0) is causing the issue, seems to be overriding the calculation of the offset value with '50'

As a workaround i have created a handler to grab that calculated value and assign it after connecting the handler

protected override void ConnectHandler(MauiSwipeRefreshLayout platformView)
{

        int deviceOffset = platformView?.ProgressViewEndOffset ?? 0;

        base.ConnectHandler(platformView);

        platformView?.SetProgressViewEndTarget(true, deviceOffset);
}

Hope it helps.

**Comment by KarsaOrlong1981** [comment]
Great, thank you I will try it.👌

**Comment by KarsaOrlong1981** [comment]
@andyx48 Perfect, your workaround solve that issue for now. I only have to add this handler to Android, configure it at MauiProgramm and boom.


![works](https://github.com/dotnet/maui/assets/69248804/65af9f4e-0a1b-44bb-938a-1269c292a050)

```
namespace DragAndDropCollectionView.Droid   
{
    public class CustomRefreshViewHandler : RefreshViewHandler
    {
        protected override void ConnectHandler(MauiSwipeRefreshLayout platformView)
        {

            int deviceOffset = platformView?.ProgressViewEndOffset ?? 0;

            base.ConnectHandler(platformView);

            platformView?.SetProgressViewEndTarget(true, deviceOffset);
        }
    }
}

                .ConfigureMauiHandlers(handlers =>
                {
#if __ANDROID__
                    handlers.AddHandler(typeof(RefreshView), typeof(Droid.CustomRefreshViewHandler));
#endif

                });
```

**Comment by jsuarezruiz** [comment]
The cause of the issue is the same here https://github.com/dotnet/maui/issues/17592

**Comment by samhouts** [comment]
Confirmed that this regressed between [8.0.0-preview.7.8842](https://github.com/dotnet/maui/releases/tag/8.0.0-preview.7.8842) and [8.0.0-rc.1.9171](https://github.com/dotnet/maui/releases/tag/8.0.0-rc.1.9171). https://github.com/dotnet/maui/pull/17080 looks sus.

**Comment by Dreamescaper** [comment]
@PureWeen 
So it won't be fixed for .NET 8 GA? I'm wondering why? Is it considered to be a low priority?
I mean, it probably affect almost every app...

**Comment by w3ori** [comment]
Same problem with released 8.0
https://github.com/dotnet/maui/issues/18987

**Comment by CorentinSwiss4** [comment]
> @andyx48 Perfect, your workaround solve that issue for now. I only have to add this handler to Android, configure it at MauiProgramm and boom.
> 
> ![works](https://private-user-images.githubusercontent.com/69248804/270349433-65af9f4e-0a1b-44bb-938a-1269c292a050.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDU5MzkxNjgsIm5iZiI6MTcwNTkzODg2OCwicGF0aCI6Ii82OTI0ODgwNC8yNzAzNDk0MzMtNjVhZjlmNGUtMGExYi00NGJiLTkzOGEtMTI2OWMyOTJhMDUwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTIyVDE1NTQyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA1NDk2Yzk1NThmZTkwNWFmMjMyY2ZhMzBjNzMxNzgyZjdjY2FhOWRhNjM3Y2E3NDgyZTkzOTczMjZkYjUwYWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2px_We2j8YYkDKf3NV9RwOQPMztXv86BqfQh9n0FGPI)
> 
> ```
> namespace DragAndDropCollectionView.Droid   
> {
>     public class CustomRefreshViewHandler : RefreshViewHandler
>     {
>         protected override void ConnectHandler(MauiSwipeRefreshLayout platformView)
>         {
> 
>             int deviceOffset = platformView?.ProgressViewEndOffset ?? 0;
> 
>             base.ConnectHandler(platformView);
> 
>             platformView?.SetProgressViewEndTarget(true, deviceOffset);
>         }
>     }
> }
> 
>                 .ConfigureMauiHandlers(handlers =>
>                 {
> #if __ANDROID__
>                     handlers.AddHandler(typeof(RefreshView), typeof(Droid.CustomRefreshViewHandler));
> #endif
> 
>                 });
> ```

Thanks a lot !
